internal/app/components/timelines/timelines.go

```diff
@@ -22,12 +22,23 @@ func newTimelineId() TimelineId {
 	return TimelineId(xid.New().String())
 }
 
-// TODO: use cr parameter when creating Timeline
-func NewTimeline(name TimelineName, cr *courses.CourseRecords) *Timeline {
+func NewTimeline(name TimelineName, courseHistory []*terms.TermRecord) *Timeline {
+	var historyCopy []*terms.TermRecord
+	for _, tr := range courseHistory {
+		uwTermId := int(tr.Term.Season) + (int(tr.Term.Year)-1900)*10
+		newRecord := terms.NewTermRecord(tr.Term.Name, uwTermId)
+		for _, cr := range tr.CourseRecords {
+			copyTime := *cr.CompletionDate
+			tempCourseRecord := cr
```

Comment:
is it really deep copying because cr is a pointer to a CourseRecord, which is in courseHistory

Reply:
True. I will dereference cr so that I would be copying the actual data structure.